src/velangtemp_ros_publisher.cpp

```diff
@@ -18,71 +18,56 @@
+    m_publisher = node->create_publisher<imu_ros2::msg::VelAngTempData>("velangtempdata", 10);
 }
 
-void AdiImuRosPublisher::setMessageProvider(AdiImuDataProviderInterface *dataProvider)
+void VelAngTempRosPublisher::setMessageProvider(VelAngTempDataProviderInterface *dataProvider)
 {
     m_dataProvider = dataProvider;
 }
 
-void AdiImuRosPublisher::run()
+void VelAngTempRosPublisher::run()
 {
     std::thread::id this_id = std::this_thread::get_id();
     std::cout << "thread " << this_id << " started...\n";
     RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "startThread: '%d'", this_id);
 
-    typedef std::chrono::system_clock::time_point timenow;
-    timenow started;
-    timenow done;
     bool success = false;
 
     rclcpp::WallRate loopRate(0.1);
 
     while (rclcpp::ok()) {
 
-       //std::thread::id this_id = std::this_thread::get_id();
-        //std::cout << "thread " << this_id << " running...\n";
-        //RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "running: '%d'", this_id);
-
         int32_t operation_mode = m_node->get_parameter("operation_mode").get_parameter_value().get<int32_t>();
 
         switch(operation_mode) {
         case DEVICE_CONTINUOUS_SAMPLING_MODE:
-            started = std::chrono::high_resolution_clock::now();
             success = false;
             m_message = m_dataProvider->getData(success);
-            done = std::chrono::high_resolution_clock::now();
 
-            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu acceleration data: '%f' '%f' '%f'",
-                        m_message.accel.x, m_message.accel.y, m_message.accel.z);
-            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu gyroscope data: '%f' '%f' '%f'",
-                        m_message.gyro.x, m_message.gyro.y, m_message.gyro.z);
-            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu delta velocity data: '%f' '%f' '%f'",
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_velangtemp"), "Publishing delta velocity data: '%f' '%f' '%f'",
                         m_message.delta_vel.x, m_message.delta_vel.y, m_message.delta_vel.z);
-            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu delta angle data: '%f' '%f' '%f'",
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_velangtemp"), "Publishing delta angle data: '%f' '%f' '%f'",
                         m_message.delta_angle.x, m_message.delta_angle.y, m_message.delta_angle.z);
-            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu temperature data: '%f' ",
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_velangtemp"), "Publishing temperature data: '%f' ",
                         m_message.temp);
-            RCLCPP_INFO(rclcpp::get_logger("rclcpp_adiimu"), "Publishing adi imu time %d us ",
-                        std::chrono::duration_cast<std::chrono::microseconds>(done-started).count());
 
             if(success)
```

Comment:
This is just a small suggestion, but maybe checking the success variable might be useful before logging the message (and moving the Logging lines within the if condition), this way it won't log random values